rest/user_api_test.go

```diff
@@ -1275,6 +1275,48 @@ func TestGetUserCollectionAccess(t *testing.T) {
 	RequireStatus(t, putResponse, 404)
 }
 
+// TestPutUserUnsetAdminChannels ensures that a PUT on the /_user/... endpoint with `null` admin_channels can be used to unset channels.
+// Repros CBG-3610, a regression introduced in 3.1.0
+func TestPutUserUnsetAdminChannels(t *testing.T) {
+	rt := NewRestTester(t, nil)
```

Comment:
Is there much/any difference?

I'm seeing relatively few `NewRestTesterPersistentConfig` in comparison to `NewRestTester` so want to understand why I should not use it.

Reply:
I think we should change `NewRestTester` but I created a helper function to make sure that we use it in new tests. It is a bigger refactoring to make all `RestTester` use persistent config and there are two variants we care about that make it hard to refactor.

1. use persistent config, and have RestTester create the database for you. This is what non persistent config `NewRestTester` does now.
2. places that actually explicitly specify `RestTesterConfig{PersistentConfig: true}` and then manually create a databases. These can't be modified.

I created this helper function recently to make it easier for new tests to use persistent config without undergoing this refactoring. For this test it doesn't matter at all, because there's no modifications to database config but I thought I'd use this opportunity to try to start a new pattern. When we change the defaults for `RestTester` it might be easier change the code that we already know works.

We don't need to change this location as is.